src/telemetry.ts

```diff
@@ -141,18 +141,19 @@ export class TelemetryWorker {
     }
 
     try {
-      return await Promise.resolve(callback.apply(
-          null, [context, outputChannel, telemetryContext, ...commandArgs]));
-      // return await callback(context, outputChannel, telemetryContext,
-      // ...commandArgs);
+      return await callback(
+          context, outputChannel, telemetryContext, ...commandArgs);
     } catch (error) {
       telemetryContext.properties.errorMessage = error.message;
+      let isPopupErrorMsg = true;
       if (error instanceof CancelOperationError) {
         telemetryContext.properties.result = TelemetryResult.Cancelled;
+        isPopupErrorMsg = false;
       } else {
         telemetryContext.properties.result = TelemetryResult.Failed;
       }
-      ExceptionHelper.logError(outputChannel, error, true);
+      telemetryContext.properties.errorMessage = error.message;
+      ExceptionHelper.logError(outputChannel, error, isPopupErrorMsg);

```

Comment:
It looks like CancelOperationError will show in output channel, do we still need to output message for CancelOperationError again in other function? e.g. CodeGeneratorCore

Reply:
All cancel operation error print log in channel through this function, including CodeGeneratorCore.